sdv/_utils.py

```diff
@@ -301,63 +307,81 @@ def _check_is_lower_version(current_version, synthesizer_version):
                 return False
 
-            if synth_v > current_v:
+            if compare_operator(synth_v, current_v):
                 return True
+
         except Exception:
             pass
 
     return False
 
 
-def check_synthesizer_version(synthesizer):
+def check_synthesizer_version(synthesizer, is_fit_method=False, compare_operator=operator.gt):
     """Check if the current synthesizer version is greater than the package version.
 
     Args:
         synthesizer (BaseSynthesizer or BaseMultiTableSynthesizer):
             An SDV model instance to check versions against.
+        is_fit_method (bool):
+            Whether or not this function is being called by a ``fit`` function.
+        compare_operator (operator):
+            Operator function to evaluate with. Defaults to ``operator.gt``.
 
     Raises:
         VersionError:
             If the current version of the software is lower than the synthesizer's version.
     """
     current_public_version = getattr(version, 'public', None)
     current_enterprise_version = getattr(version, 'enterprise', None)
-    if synthesizer._fitted:
-        downgrade_message = 'Downgrading your SDV version is not supported.'
-        fit_public_version = getattr(synthesizer, '_fitted_sdv_version', None)
-        fit_enterprise_version = getattr(synthesizer, '_fitted_sdv_enterprise_version', None)
-
-        is_public_lower = _check_is_lower_version(current_public_version, fit_public_version)
-        is_enterprise_lower = False
-        if None not in (current_enterprise_version, fit_enterprise_version):
-            is_enterprise_lower = _check_is_lower_version(
-                current_enterprise_version,
-                fit_enterprise_version
-            )
+    static_message = 'Downgrading your SDV version is not supported.'
+    if is_fit_method:
+        static_message = (
+            'Fitting this synthesizer again is not supported. '
+            'Please create a new synthesizer.'
+        )
 
-        if is_public_lower and is_enterprise_lower:
-            raise VersionError(
-                f'You are currently on SDV version {current_public_version} and SDV Enterprise '
-                f'version {current_enterprise_version} but this '
-                f'synthesizer was created on SDV version {fit_public_version} and SDV '
-                f'Enterprise version {fit_enterprise_version}. {downgrade_message}'
-            )
+    fit_public_version = getattr(synthesizer, '_fitted_sdv_version', None)
+    fit_enterprise_version = getattr(synthesizer, '_fitted_sdv_enterprise_version', None)
 
-        if is_public_lower:
-            raise VersionError(
-                f'You are currently on SDV version {current_public_version} but this '
-                f'synthesizer was created on version {fit_public_version}. {downgrade_message}'
-            )
+    is_public_lower = False
+    if None not in (current_public_version, fit_public_version):
```

Comment:
why not have compare_versions return False if either one is None

Reply:
I think that by the `try except` it should return false actually.